libcore/src/SimulationHelper.cpp

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (c) 2020 Forschungszentrum Jülich GmbH. All rights reserved.
+ *
+ * \section License
+ * This file is part of JuPedSim.
+ *
+ * JuPedSim is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * JuPedSim is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with JuPedSim. If not, see <http://www.gnu.org/licenses/>.
+ **/
+#include "SimulationHelper.h"
+
+#include "general/Logger.h"
+#include "geometry/Room.h"
+#include "geometry/SubRoom.h"
+
+#include <algorithm>
+
+PedRelocation
+SimulationHelper::UpdatePedestrianRoomInformation(const Building & building, Pedestrian & ped)
+{
+    // No relocation needed, ped is in its assigned room/subroom
+    // TODO add check if room/subroom really exist
+    auto oldSubRoom = building.GetRoom(ped.GetRoomID())->GetSubRoom(ped.GetSubRoomID());
+    if(oldSubRoom->IsInSubRoom(&ped)) {
+        return PedRelocation::NOT_NEEDED;
+    }
+
+    // relocation needed, look if pedestrian is in one of the neighboring rooms/subrooms
+    // get all connected directly connected rooms/subrooms
+    // TODO check for closed doors?
+    auto subroomsConnected = oldSubRoom->GetNeighbors();
+    auto currentSubRoom    = std::find_if(
+        std::begin(subroomsConnected),
+        std::end(subroomsConnected),
+        [&ped](auto const & subroom) -> bool { return subroom->IsInSubRoom(ped.GetPos()); });
+
+    // pedestrian is in one of the neighboring rooms/subrooms
+    if(currentSubRoom != subroomsConnected.end()) {
+        ped.UpdateRoom((*currentSubRoom)->GetRoomID(), (*currentSubRoom)->GetSubRoomID());
+        ped.SetSubRoomUID((*currentSubRoom)->GetUID());
+        return PedRelocation::SUCCESSFUL;
+    } else {
+        ped.UpdateRoom(-1, -1);
+        ped.SetSubRoomUID(-1);
+        return PedRelocation::FAILED;
+    }
+}
+
+std::vector<Pedestrian *> SimulationHelper::FindPedsReachedFinalGoal(
```

Comment:
We should be more consistent on Ped vs Pedestrian and singular and plural.

Reply:
- Pedestrian: In function name when only used on single pedestrian
- Pedestrians: In function name when used on multiple/all pedestrian
- ped: In parameter name if function works on single pedestrian
- peds: In parameter name if function works on multiple pedestrians